cmd/decrypt.go

```diff
@@ -17,24 +19,38 @@ var decryptCmd = &cobra.Command{
 	RunE:    DecryptCommand,
 }
 
+var allowFail bool
+
 func init() {
 	RootCmd.AddCommand(decryptCmd)
+	decryptCmd.Flags().BoolVarP(&allowFail, "allow-fail", "a", false, "Do not fail if not all files can be decrypted")
 }
 
 // DecryptCommand decrypts files
 func DecryptCommand(cmd *cobra.Command, args []string) error {
+	initConfig()
+
 	filesToDecrypt, err := fileutil.SomeOrAllFiles(args, sopsConfig.EncryptedFiles)
 	if err != nil {
 		return err
 	}
 
 	//decrypt all the files
+	errCount := 0
 	for _, f := range filesToDecrypt {
 		err := execwrap.DecryptFile(f)
 		if err != nil {
-			return err
+			if allowFail {
+				errCount++
+			} else {
+				return err
+			}
 		}
 	}
 
+	if errCount == len(filesToDecrypt) {
+		return fmt.Errorf("all files failed to decrypt")
```

Comment:
So even with `--allow-fail` this will be non-zero if all files fail to decrypt? It feels like "allow fail" should never return a failure exit code.

I see where you're coming from based on the context of the original discussion (allow staging file to fail to decrypt in production), but I think the all failure scenario would be better handled in application code, maybe?